### ISSUE TITLE
dune syntax errors are cryptic, especially when porting from jbuild syntax

### ISSUE THREAD

**Issue by andreypopp** [ocaml/dune#981]
I've tried to port a project from jbuilder to dune and one of the first errors I found was:
```
info esy build 0.1.27
File "bin/dune", line 2, characters 2-14:
Error: Atom expected
```
with the following `bin/dune`:
```
(executable (
  (name hello)
  (public_name hello)
  (libraries (lib))
))
```

As I'm following dune development I've started to remove parens as I saw a PR which simplifies the syntax in dune and eventually succeeded.

I suspect that others (who don't pay detailed attention) will also struggle with that — could it be made so that syntax with extra parens is still parsed but emits warnings? Or at least errors with a more detailed description of an error.

**Comment by rgrinberg** [comment]
What I suspect is possible is trying to re-parse a stanza with the jbuilder syntax after failing in dune mode. If we succeed parsing using jbuild, we can emit a better error message that hints at what needs to be done. @diml is that feasible/desirable?

**Comment by diml** [comment]
I misunderstood your suggestion. Yes, that's a good idea actually, it should work nicely

**Comment by diml** [comment]
I implemented something like this in #983

**Comment by diml** [comment]
Things should be better now, please let us know if you find more error messages that could be improved.
